students/luftsmeerflier/lesson03/slicing_lab.py

```diff
@@ -3,48 +3,29 @@
 # with the first and last items exchanged
 
 def exchange_first_last(seq):	
-	def switch(input_):
-		output = input_
+    def switch(input_):
+	    output = input_
 		reverse = input_[::-1]
-		output[0:1] = reverse[0:1]
-		output[-1] = reverse[-1]
-		return output
-		
-	if type(seq) == tuple:
-		return tuple(switch(list(seq)))
-	elif type(seq) == str:
-		return ''.join(switch(list(seq)))
-	else:
-		print('Please input a string or tuple')
+		if type(input_) == str:
+			return reverse[0:1] + output[1:-1] + reverse[0-1]
+		else:
+			output = list(output)
+			output[0:1] = reverse[0:1]
+			output[-1] = reverse[-1]
+			return tuple(output)
+
+	return switch(seq)
```

Comment:
same comment as last time, you want to use slicing only and this just takes one line of code:
```
return seq[-1:] + seq[1:-1] + seq[:1]

```


you can see below that slice syntax will return original data structure:
```
>>> seq = (1, 2, 3, 4)
>>> seq[-1:] + seq[1:-1] + seq[:1]
(4, 2, 3, 1)
>>> seq[:1]
(1,)
>>>
```